frontends/web/src/routes/account/receive/receive.jsx

```diff
@@ -142,36 +161,77 @@ export default class Receive extends Component {
         } else if (account.coinCode === 'eth' || account.coinCode === 'teth' || account.coinCode === 'reth') {
             uriPrefix = '';
         }
+        // enable copying only after verification has been invoked if verification is possible and not optional.
+        const forceVerification = secureOutput.hasSecureOutput && !secureOutput.optional;
+        let enableCopy = !forceVerification || verified;
+        let address;
+        if (receiveAddresses) {
+            address = receiveAddresses[activeIndex].address;
+            if (!enableCopy && !verifying) {
+                address = address.substring(0, 8) + '...';
+            }
+        }
         const content = receiveAddresses ? (
             <div>
-                <QRCode data={uriPrefix + receiveAddresses[activeIndex].address} />
-                <CopyableInput value={receiveAddresses[activeIndex].address} />
-                <div class={['flex flex-row flex-center flex-items-center', style.labels].join(' ')}>
+                <div class={style.qrCodeContainer}>
+                    <QRCode data={enableCopy ? uriPrefix + address : undefined} />
+                </div>
+                <div class={['flex flex-row flex-between flex-items-center', style.labels].join(' ')}>
                     {
                         receiveAddresses.length > 1 && (
-                            <Button
-                                transparent
+                            <a
+                                href="!#"
+                                className={['flex flex-row flex-items-center', style.previous].join(' ')}
                                 disabled={verifying}
                                 onClick={this.previous}>
-                                <img src={ArrowLeft} class={style.arrowLeft} />
+                                <img src={ArrowLeft} />
                                 {t('button.previous')}
-                            </Button>
+                            </a>
                         )
                     }
                     <p class={style.label}>{t('receive.label')} { receiveAddresses.length > 1 ? `(${activeIndex + 1}/${receiveAddresses.length})` : ''}</p>
                     {
                         receiveAddresses.length > 1 && (
-                            <Button
-                                transparent
+                            <a
+                                href="!#"
```

Comment:
what is `!#`? (I've only seen `#` before)

Reply:
typo.. lol although there is such thing as a hashbang (#!), where it means something different. fixed.